kart/spatial_filters.py

```diff
@@ -333,6 +335,25 @@ def split_file(cls, contents):
         return [p.strip() for p in parts]
 
 
+class MatchResult(Enum):
+    """Stores a bit more detail about whether a feature matches or doesn't match the spatial filter."""
+
+    # The given feature...
+    # Doesn't match any spatial filter because it is null / None / non-existent:
+    IS_NONE = auto()
+    # Isn't present locally, but is promised. From this we infer it doesn't match the repo's current spatial filter:
+    IS_PROMISED = auto()
+    # Exists and is present and does NOT match the given spatial filter:
+    NON_MATCHING = auto()
+    # Exists and is present and DOES match the given spatial filter.
+    # This includes the case where a feature matches the "match-all" spatial-filter, which matches everything,
+    # and it includes the case where a feature has no geometry (and so matches all spatial filters).
+    MATCHING = auto()
```

Comment:
these are active voice, the promised is passive... maybe `NOT_MATCHED` & `MATCHED`?

Alternatively I guess non-matching could be is-outside/excluded/is-out?